rpcserver.go

```diff
@@ -7074,8 +7074,24 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 		// amount based on the asset-to-BTC conversion rate.
 		sellOrder := taprpc.MarshalAcceptedSellQuote(*quote)
 
+		// paymentMaxAmt is the maximum amount that the counterparty is
+		// expected to pay. This is the amount that the invoice is
+		// asking for plus the fee limit in milli-satoshis.
+		paymentMaxAmt, _, err := r.parseRequest(pReq)
+		if err != nil {
+			return err
+		}
+
+		// Check if the payment is economical based on the quote.
```

Comment:
"Payment is economical" doesn’t seem quite right here. We’re not evaluating whether the payment is a good value or fair price relative to what’s received. Instead, we’re specifically checking for overpayment—meaning the total payment value exceeds the invoice amount.